bftengine/src/bftengine/ClientsManager.cpp

```diff
@@ -227,15 +229,16 @@ std::shared_ptr<ClientReplyMsg> ClientsManager::allocateReplyFromSavedOne(NodeId
   const uint16_t clientIdx = clientIdToIndex_.at(clientId);
   const uint32_t firstPageId = clientIdx * reservedPagesPerClient_;
   LOG_DEBUG(CL_MNGR, KVLOG(clientId, requestSeqNum, firstPageId));
-  for (int j = 0; j < maxNumOfReqsPerClient_; j++) {
-    stateTransfer_->loadReservedPage(resPageOffset() + firstPageId + j, sizeOfReservedPage_, scratchPage_.data());
+  for (uint16_t j = 0; j < maxNumOfReqsPerClient_; j++) {
+    stateTransfer_->loadReservedPage(
+        resPageOffset() + firstPageId + (j * singleReplyMaxNumOfPages_), sizeOfReservedPage_, scratchPage_.data());
 
     ClientReplyMsgHeader* replyHeader = (ClientReplyMsgHeader*)scratchPage_.data();
+    if (replyHeader->reqSeqNum != requestSeqNum) continue;
     ConcordAssert(replyHeader->msgType == MsgCode::ClientReply);
     ConcordAssert(replyHeader->currentPrimaryId == 0);
     ConcordAssert(replyHeader->replyLength > 0);
     ConcordAssert(replyHeader->replyLength + sizeof(ClientReplyMsgHeader) <= maxReplySize_);
-    if (replyHeader->reqSeqNum != requestSeqNum) continue;
```

Comment:
Why this line has gone?

Reply:
Just moved before the asserts, we want to first check that this is the right request.

Reply:
From this diif it looked like removed